core/src/main/java/org/jboss/intersmash/util/CommandLineBasedKeystoreGenerator.java

```diff
@@ -0,0 +1,347 @@
+						"-out", hostname + ".cer",
+						"-days", "365",
+						"-CAcreateserial",
+						"-passin",
+						"pass:" + password,
+						"-extfile",
+						keyAlias + ".extensions",
+						"-extensions",
+						"req_ext");
+			} catch (IOException e) {
+				throw new RuntimeException(e);
+			}
+		} else {
+			processCall(caDir, "openssl",
+					"x509",
+					"-req",
+					"-CA",
+					CA_CERTIFICATE_PEM_FILE_NAME,
+					"-CAkey",
+					"ca-key.pem",
+					"-in",
+					hostname + ".csr",
+					"-out",
+					hostname + ".cer",
+					"-days",
+					"365",
+					"-CAcreateserial",
+					"-passin",
+					"pass:" + password);
+		}
+		processCall(caDir, "keytool",
+				"-import",
+				"-noprompt",
+				"-keystore",
+				keystore,
+				"-file",
+				CA_CERTIFICATE_PEM_FILE_NAME,
+				"-alias",
+				INTERSMASH_CA,
+				"-storepass",
+				password);
+		processCall(caDir, "keytool",
+				"-import",
+				"-keystore",
+				keystore,
+				"-file",
+				hostname + ".cer",
+				"-alias",
+				keyAlias,
+				"-storepass", password);
+		if (deleteCaFromKeyStore) {
+			processCall(caDir, "keytool",
+					"-delete",
+					"-noprompt",
+					"-alias", INTERSMASH_CA,
+					"-keystore",
+					keystore,
+					"-storepass",
+					password);
+		}
+		return caDir.resolve(keystore);
+	}
+
+	public static GeneratedPaths generateCerts(String hostname) {
+		return generateCerts(hostname, null);
+	}
+
+	public static GeneratedPaths generateCerts(String hostname, String[] alternativeHostnames) {
+		final String keystore = hostname + KEYSTORE_FILE_EXTENSION;
+		generateKeystore(hostname, alternativeHostnames);
+		// export cert as CN.keystore.pem
+		processCall(caDir, "keytool",
+				"-exportcert",
+				"-rfc",
+				"-keystore",
+				keystore,
+				"-alias",
+				hostname,
+				"-storepass",
+				password,
+				"-file",
+				keystore + ".pem");
+		// convert to CN.keystore.keywithattrs.pem
+		processCall(caDir, "openssl",
+				"pkcs12",
+				"-in",
+				keystore,
+				"-nodes",
+				"-nocerts",
+				"-out",
+				keystore + ".keywithattrs.pem",
+				"-passin",
+				"pass:" + password);
+		// Remove the Bag attributes to CN.keystore.key.pem
+		processCall(caDir, "openssl",
+				"rsa",
+				"-in",
+				keystore + ".keywithattrs.pem",
+				"-out",
+				keystore + ".key.pem");
+		return new GeneratedPaths(
+				caDir.resolve(CA_CERTIFICATE_PEM_FILE_NAME),
+				caDir.resolve(TRUSTSTORE_FILE_NAME),
+				caDir.resolve(keystore),
+				caDir.resolve(keystore + ".key.pem"),
+				caDir.resolve(keystore + ".pem"));
+	}
+
+	private static String getOpenShiftHostAndPort() {
+		String server = OpenShiftConfig.url().replaceFirst("https://", "");
+		server = server.endsWith("/") ? server.substring(0, server.length() - 1) : server;
+		server = server.contains(":") ? server : server + ":443";
+		return server;
+	}
+
+	private static String getSplitCommandNameBasedOnOeratingSystem() {
+		return System.getProperty("os.name").toLowerCase().startsWith("mac") ? "gcsplit" : "csplit";
```

Comment:
Is mac supported by other parts of intersmash as well?

Reply:
No, there are no other places where we take it into account.
That must be a leftover of early implementation of this utility class.
I might file an issue for removal, if you'd be okay with that, or remove it now, although this would mean another "costly" integration-tests round.
WDYT @marekkopecky ?

Reply:
+1, let's merge it now, the issue for the future would be great.

Reply:
Thanks @marekkopecky - here's the issue tracker: https://github.com/Intersmash/intersmash/issues/198